pkg/binding/backend/interactive.go

```diff
@@ -98,6 +102,38 @@ loop:
 	return selectedName, selectedGVK, nil
 }
 
+// SelectNamespaceForServiceInstances returns the namespace which services instances should be listed from.
+// An empty return value means that service instances will be listed from the current namespace.
+// Otherwise, they will be listed from the specified namespace.
+func (o *InteractiveBackend) SelectNamespaceForServiceInstances(_ map[string]string, nsListProvider func() ([]string, error)) (string, error) {
+	option, err := o.askerClient.SelectNamespaceListOption()
+	if err != nil {
+		return "", err
+	}
+
+	switch option {
+	case asker.CurrentNamespace:
+		return "", nil
+	case asker.AllAccessibleNamespaces:
+		klog.V(2).Infof("Listing all projects/namespaces...")
+		var nsList []string
+		nsList, err = nsListProvider()
+		if err != nil {
+			return "", err
+		}
+		sort.Strings(nsList)
+		klog.V(4).Infof("all accessible namespaces: %v", nsList)
+		if len(nsList) == 0 {
```

Comment:
Out of curiosity, when is this condition going to be satisfied? The user would always have the current working namespace in the `nsList`, no?

Reply:
If the user does not have the permission to list the namespace, this condition will become true.

Reply:
Thanks, @valaparthvi. That didn't occur to me.

Reply:
Indeed, this can happen because of RBAC issues. In this case, the user will be prompted to manually enter a namespace (or press Enter to select the current one).
I wanted to test this scenario after reading a difference in behavior between listing all projects in OpenShift and listing all namespaces in Kubernetes. In OpenShift, it returns the list of all projects the user really has access to. But in Kubernetes, by design, it returns an error if the user does not have permission to list namespaces.

See message of commit [88e6902a0819256a9508c4b7f30beba6b04dacef](https://github.com/redhat-developer/odo/pull/5936/commits/88e6902a0819256a9508c4b7f30beba6b04dacef) for more details.